### PR TITLE
feat: Add new pattern for EKS cross account communication using VPC lattice

### PR DESCRIPTION
**Pull request by allamand**
# Description

This PR introduce a new pattern that show case how to leverage VPC lattice for cross-cluster communication, in a secure manner relying on AWS Private CA and AWS Certificate Manager on a private Route 53 hosted zone.
The pattern demonstrate how to leverage EKS Pod Identity Session tags to do service level authorization with ABAC verifications.

### Motivation and Context

- This pattern was asked by Networking team.

### How was this change tested?

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

### CODE REVIEW

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/templates/app-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  template:
+    metadata:
+      annotations:
+       vpc-lattices-svcs.amazonaws.com/agent-inject: "true"
+      labels:
+        app: {{ .Release.Name }}-{{ .Values.version }}
+    spec:
+      containers:
+      - name: {{ .Release.Name }}-{{ .Values.version }}
+        image: public.ecr.aws/seb-demo/http-server:latest
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/templates/app-deploy.yaml:22]: Can we use some public image like [this](https://github.com/kubernetes-sigs/aws-load-balancer-controller/blame/e5d625f96415fd44e6399e9c75e2bd985f5a2288/docs/examples/echoservice/echoserver-deployment.yaml#L17) instead?

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/values.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/values.yaml:1]:
```suggestion
```

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+apiVersion: kyverno.io/v1
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml:1]: This needs more explanation in the documentation. Its an important detail that is not getting surfaced up properly.

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      mutate:
+        patchStrategicMerge:
+          spec:
+            template:
+              metadata:
+                annotations:
+                  (vpc-lattices-svcs.amazonaws.com/agent-inject): "true"
+              spec:
+                initContainers: # IPTables rules are updated in init container
+                  - image: public.ecr.aws/d2c6w7a3/iptables
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml:24]: Can we use a public image? Such as https://hub.docker.com/r/istio/iptables?

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                        iptables -t nat -N EGRESS_PROXY;
+                        iptables -t nat -A OUTPUT -p tcp -d 169.254.171.0/24 -j EGRESS_PROXY;
+                        iptables -t nat -A EGRESS_PROXY -m owner --uid-owner 101 -j RETURN;
+                        iptables -t nat -A EGRESS_PROXY -p tcp -j REDIRECT --to-ports 8080;
+                containers:
+                  - name: sigv4proxy
+                    env:
+                      - name: AWS_REGION
+                        value: "{{ .Values.region }}"
+                    image: public.ecr.aws/seb-demo/aws-sigv4-proxy:latest
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml:43]: https://gallery.ecr.aws/aws-observability/aws-sigv4-proxy  perhaps?

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf [hunk 6]
```diff
@@ -0,0 +1,270 @@
+################################################################################
+# Cluster
+################################################################################
+
+module "eks" {
+  source  = "terraform-aws-modules/eks/aws"
+  version = "~> 20.8"
+
+  cluster_name                   = local.name
+  cluster_version                = "1.29"
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:10]:
```suggestion
  cluster_version                = "1.30"
```

?

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf [hunk 7]
```diff
@@ -0,0 +1,270 @@
...
+        value = module.vpc.vpc_id
+      },
+      {
+        name  = "defaultServiceNetwork"
+        value = "lattice-gateway" # Specify this parameter to create a default VPC lattice service
+      },
+      # {
+      #   name  = "latticeEndpoint"
+      #   value = "https://vpc-lattice.${local.region}.amazonaws.com"
+      # }
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:91]: Remove if not needed.

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/values.yaml [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+awsAccountID: to-replace
+region: to-replace
+vpc1ID: to-replace
+vpc2ID: to-replace
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/values.yaml:5]:
```suggestion
```

Comment (askulkarni2) [reply patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/values.yaml:5]: I don't think you are using these.

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/values.yaml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+awsAccountID: to-replace
+region: to-replace
+vpc1ID: to-replace
+vpc2ID: to-replace
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/values.yaml:4]:
```suggestion
```

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf [hunk 10]
```diff
@@ -0,0 +1,270 @@
...
+    <<EOF
+    awsAccountID: "${data.aws_caller_identity.current.account_id}"
+    region: "${local.region}"
+    version : "v1"
+
+    allowedCluster: ${terraform.workspace == "cluster1" ? "eks-cluster2" : "eks-cluster1"}
+    allowedNamespace: "apps"
+
+    vpc1ID: ${module.vpc.vpc_id}
+    vpc2ID: to-replace
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:200]:
```suggestion
```

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf [hunk 11]
```diff
@@ -0,0 +1,270 @@
...
+  create_namespace = true
+  namespace        = "lattice-gateway"
+  force_update     = true
+
+  #replace = true # This will force a re-deployment
+
+  values = [
+    <<EOF
+    awsAccountID: "${data.aws_caller_identity.current.account_id}"
+    region: "${local.region}"
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:193]:
```suggestion
```

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf [hunk 12]
```diff
@@ -0,0 +1,270 @@
...
+  create_namespace = true
+  namespace        = local.app_namespace
+  force_update     = true
+
+  #replace = true # This will force a re-deployment
+
+  values = [
+    <<EOF
+    awsAccountID: "${data.aws_caller_identity.current.account_id}"
+    region: "${local.region}"
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:227]:
```suggestion
```

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf [hunk 13]
```diff
@@ -0,0 +1,270 @@
...
+    version : "v1"
+
+    allowedCluster: ${terraform.workspace == "cluster1" ? "eks-cluster2" : "eks-cluster1"}
+    allowedNamespace: "apps"
+
+    vpc1ID: ${module.vpc.vpc_id}
+    vpc2ID: to-replace
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:234]:
```suggestion
```

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/logs.txt [hunk 14]
```diff
@@ -0,0 +1,761 @@
+[0m[1mmodule.eks.module.eks_managed_node_group["initial"].data.aws_iam_policy_document.assume_role_policy[0]: Reading...[0m[0m
```
Comment (askulkarni2) [inline patterns/vpc-lattice/cross-cluster-pod-communication/cluster/logs.txt:1]: Delete this

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/templates/app-deploy.yaml (hunk 1 above)
Comment (allamand) [reply patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/templates/app-deploy.yaml:22]: not sure, the code to build the container is public : https://github.com/aws-samples/amazon-eks-security-immersion-day/blob/mainline/docker/http-server/launch_app.sh and I use a script to configure the CA at runtime

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf (hunk 11 above)
Comment (allamand) [reply patterns/vpc-lattice/cross-cluster-pod-communication/cluster/eks.tf:193]: why you suggest to remove region ? this is used in the helm chart to configure vpc lattice envoy container

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml (hunk 3 above)
Comment (allamand) [reply patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml:1]: I plan to link this also to the associated blog post. I added more info on it but want to keep REadme light

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml (hunk 4 above)
Comment (allamand) [reply patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/platform/templates/kyverno-cluster-policy.yaml:24]: I missed this comment, I will try with public image

File: patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/templates/app-deploy.yaml (hunk 1 above)
Comment (allamand) [reply patterns/vpc-lattice/cross-cluster-pod-communication/cluster/charts/demo/templates/app-deploy.yaml:22]: I put the link to the Dockerfile used

**Review by askulkarni2 — APPROVED** [review]
LGTM!

**Pull request merged**